pkgs/development/libraries/ncurses/default.nix

```
@@ -24,14 +24,22 @@ stdenv.mkDerivation (finalAttrs: {
   outputs = [ "out" "dev" "man" ];
   setOutputFlags = false; # some aren't supported
 
-  configureFlags = [
+  configureFlags = let
+    lld17Plus = stdenv.cc.bintools.isLLVM
+      && lib.versionAtLeast stdenv.cc.bintools.version "17";
+  in [
     (lib.withFeature (!enableStatic) "shared")
     "--without-debug"
     "--enable-pc-files"
     "--enable-symlinks"
     "--with-manpage-format=normal"
     "--disable-stripping"
     "--with-versioned-syms"
+  ] ++ lib.optionals lld17Plus [
```

```suggestion
  ] ++ lib.optionals (stdenv.cc.bintools.isLLVM && lib.versionAtLeast stdenv.cc.bintools.version "17") [
```
the other conditions in that file don't do that either.

---

I think it would be good to move this to the end too, so all the -- flags are together, followed by environment variable overrides.  Otherwise it's difficult to see whether a variable has already been set.